sql/mozfun/iap/derive_apple_subscription_interval/udf.sql

```diff
@@ -0,0 +1,117 @@
+CREATE OR REPLACE FUNCTION iap.derive_apple_subscription_interval(
+  purchase_datetime_pst DATETIME,
+  expires_datetime_pst DATETIME
+) AS (
+  CASE
+  WHEN
+    DATETIME_ADD(purchase_datetime_pst, INTERVAL 1 YEAR) = expires_datetime_pst
+  THEN
+    "year"
+  WHEN
+    DATETIME_ADD(purchase_datetime_pst, INTERVAL 1 QUARTER) = expires_datetime_pst
+  THEN
+    "quarter"
+  WHEN
+    DATETIME_ADD(purchase_datetime_pst, INTERVAL 1 MONTH) = expires_datetime_pst
+  THEN
+    "month"
+  WHEN
+    DATETIME_ADD(purchase_datetime_pst, INTERVAL 1 WEEK) = expires_datetime_pst
+  THEN
+    "week"
+  WHEN
+    DATETIME_ADD(purchase_datetime_pst, INTERVAL 1 DAY) = expires_datetime_pst
+  THEN
+    "day"
+  ELSE
+    CONCAT(CAST(DATETIME_DIFF(expires_datetime_pst, purchase_datetime_pst, DAY) AS STRING), " day")
```

Comment:
Why is there a special case for having a number of days?

Reply:
because it's a fallback, and I didn't want it to throw and error or lose information if we add something weird in the future that this doesn't recognize